cwltool/main.py

```diff
@@ -695,6 +697,18 @@ def main(argsl=None,  # type: List[str]
                 argsl = sys.argv[1:]
             args = arg_parser().parse_args(argsl)
 
+        # If On windows platform, A default Docker Container is Used if not explicitely provided by user
+        if onWindows() and not args.default_container:
+            args.default_container = "ubuntu"
+
+        # Making default Container as lowest priority, that should be used if nothing overrides it
+        if args.default_container:
```

Comment:
Actually the previously default container code was placed before job. So it was not propogating to workflow steps. I removed that and changed the position. Strangely can't find that old implementation. I guess some PR changed it. I would see it again.